crates/transaction-pool/src/traits.rs

```diff
@@ -272,6 +275,121 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + FromRecoveredTransaction {
     fn size(&self) -> usize;
 }
 
+// #[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
```

Comment:
let's not make this default, because this should only ever be initialized via the underlying tx

```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```